### PR TITLE
Also include C++ tarballs in releases

### PR DESCRIPTION
**Pull request by Luthaf**
This mean making a release is just a matter of pushing a tag to the repo. Distribution of the C++ parts of metatensor (e.g. pre-built versions in conda-forge) will then be able to use the tarball attached to the release as the sources to build.

----
:books: Documentation preview :books:: https://metatensor--386.org.readthedocs.build/en/386/

### CODE REVIEW

**Comment by Luthaf** [comment]
Ok, the comment above is not added for PR from a fork because of https://github.com/orgs/community/discussions/25220. I'll see what we can/should do about this later.

File: scripts/package-torch.sh [hunk 1]
```diff
@@ -7,17 +7,22 @@
...
-./scripts/n-commits-since-last-tag.py "metatensor-torch-v" > metatensor-torch/cmake/n_commits_since_last_tag
-tar cf "$ARCHIVE_NAME".tar metatensor-torch
-rm -f metatensor-torch/cmake/n_commits_since_last_tag
+TMP_DIR=$(mktemp -d)
+mkdir "$TMP_DIR/$ARCHIVE_NAME"
+
+cp -r "$ROOT_DIR"/metatensor-torch/* "$TMP_DIR/$ARCHIVE_NAME/"
+cp "$ROOT_DIR/LICENSE" "$TMP_DIR/$ARCHIVE_NAME"
+cp "$ROOT_DIR/AUTHORS" "$TMP_DIR/$ARCHIVE_NAME"
+./scripts/n-commits-since-last-tag.py "metatensor-torch-v" > "$TMP_DIR/$ARCHIVE_NAME/cmake/n_commits_since_last_tag"
```
Comment (agoscinski) [inline scripts/package-torch.sh:19]: maybe can you add a small comment that the produced file is used during building?

**Review by agoscinski — CHANGES_REQUESTED** [review]
Just some suggested changes. Looks good.

File: .github/workflows/build-wheels.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ name: Build Python wheels
 on:
   push:
     branches: [master]
-    tags: "*"
+    tags: ["*"]
```
[the remark this answers is not in the document]
Comment (Luthaf) [reply .github/workflows/build-wheels.yml:6]: just consistency with the docs building file which also runs on tags

**Review by agoscinski — APPROVED** [review]
Seems like there is only a problem with readthedocs, PR looks fine.

**Pull request merged**